tests/monitoring_processes/test_monitoring_critical_processes.py

```diff
@@ -0,0 +1,401 @@
+from tests.common.helpers.assertions import pytest_require
+from tests.common.helpers.constants import DEFAULT_ASIC_ID, NAMESPACE_PREFIX
+from tests.common.helpers.dut_utils import get_program_info
+from tests.common.helpers.dut_utils import get_group_program_info
+from tests.common.plugins.loganalyzer.loganalyzer import LogAnalyzer, LogAnalyzerError
+from tests.common.utilities import wait_until
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('any'),
+    pytest.mark.disable_loganalyzer
+]
+
+CONTAINER_CHECK_INTERVAL_SECS = 1
+CONTAINER_RESTART_THRESHOLD_SECS = 180
+
+
+@pytest.fixture(autouse=True, scope='module')
+def config_reload_after_tests(duthost):
+    yield
+    config_reload(duthost)
+
+
+@pytest.fixture(autouse=True, scope='module')
+def disable_and_enable_autorestart(duthost):
+    """Changes the autorestart of containers from `enabled` to `disabled` before testing.
+       and Rolls them back after testing.
+
+    Args:
+        duthost: Hostname of DUT.
+
+    Returns:
+        None.
+    """
+    containers_autorestart_states = duthost.get_container_autorestart_states()
+    disabled_autorestart_containers = []
+
+    for container_name, state in containers_autorestart_states.items():
+        if state == "enabled":
+            logger.info("Disabling the autorestart of container '{}'.".format(container_name))
+            command_disable_autorestart = "sudo config feature autorestart {} disabled".format(container_name)
+            command_output = duthost.shell(command_disable_autorestart)
+            exit_code = command_output["rc"]
+            pytest_assert(exit_code == 0, "Failed to disable the autorestart of container '{}'".format(container_name))
+            logger.info("The autorestart of container '{}' was disabled.".format(container_name))
+            disabled_autorestart_containers.append(container_name)
+
+    yield
+
+    for container_name in disabled_autorestart_containers:
+        logger.info("Enabling the autorestart of container '{}'...".format(container_name))
+        command_output = duthost.shell("sudo config feature autorestart {} enabled".format(container_name))
+        exit_code = command_output["rc"]
+        pytest_assert(exit_code == 0, "Failed to enable the autorestart of container '{}'".format(container_name))
+        logger.info("The autorestart of container '{}' is enabled.".format(container_name))
+
+
+@pytest.fixture(autouse=True, scope="module")
+def check_image_version(duthost):
+    """Skips this test if the SONiC image installed on DUT was 201911 or old version.
+
+    Args:
+        duthost: Hostname of DUT.
+
+    Returns:
+        None.
+    """
+    pytest_require(parse_version(duthost.kernel_version) > parse_version("4.9.0"),
+                   "Test was not supported for 201911 and older image versions!")
+
+
+def check_all_critical_processes_status(duthost):
+    """Post-checks the status of critical processes.
```

Comment:
This function could be called in more than a post-check scenario, if desired. Suggest rewording:

```
Determines if all critical processes are running on a DUT
```

Reply:
Updated!